aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -199,8 +212,89 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
-        let _ = (issuer_did, cred_def_id, tails_dir, max_creds, tag);
-        Err(unimplemented_method_err("credx issuer_create_and_store_revoc_reg"))
+        let issuer_did = issuer_did.to_owned().into();
+
+        let options = json!({
+            "retrieve_type": false,
+            "retrieve_value": true,
+            "retrieve_tags": false,
+        })
+        .to_string();
+
+        let mut tails_writer = TailsFileWriter::new(Some(tails_dir.to_owned()));
+
+        let cred_def_str = self
+            .wallet
+            .get_wallet_record(CATEGORY_CRED_DEF, cred_def_id, &options)
+            .await?;
+
+        let cred_def = serde_json::from_str(&cred_def_str)?;
+
+        let rev_reg_id =
+            credx::issuer::make_revocation_registry_id(&issuer_did, &cred_def, tag, RegistryType::CL_ACCUM)?;
+
+        let res_rev_reg = self
+            .wallet
+            .get_wallet_record(CATEGORY_REV_REG, &rev_reg_id.0, &options)
+            .await;
+
+        let res_rev_reg_def = self
+            .wallet
+            .get_wallet_record(CATEGORY_REV_REG_DEF, &rev_reg_id.0, &options)
+            .await;
+
+        if let (Ok(rev_reg), Ok(rev_reg_def)) = (res_rev_reg, res_rev_reg_def) {
+            return Ok((rev_reg_id.0, rev_reg, rev_reg_def));
+        }
+
+        let (rev_reg_def, rev_reg_def_priv, rev_reg, rev_reg_delta) = credx::issuer::create_revocation_registry(
+            &issuer_did,
+            &cred_def,
+            tag,
+            RegistryType::CL_ACCUM,
+            IssuanceType::ISSUANCE_BY_DEFAULT,
+            max_creds,
+            &mut tails_writer,
+        )?;
+
+        // Store stuff in wallet
+        let rev_reg_info = RevocationRegistryInfo {
+            id: rev_reg_id.clone(),
+            curr_id: 0,
+            used_ids: HashSet::new(),
+        };
+
+        let str_rev_reg_info = serde_json::to_string(&rev_reg_info)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_REV_REG_INFO, &rev_reg_id.0, &str_rev_reg_info, None)
+            .await?;
+
+        let str_rev_reg_def = serde_json::to_string(&rev_reg_def)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_REV_REG_DEF, &rev_reg_id.0, &str_rev_reg_def, None)
+            .await?;
+
+        let str_rev_reg_def_priv = serde_json::to_string(&rev_reg_def_priv)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_REV_REG_DEF_PRIV, &rev_reg_id.0, &str_rev_reg_def_priv, None)
+            .await?;
+
+        let str_rev_reg = serde_json::to_string(&rev_reg)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_REV_REG, &rev_reg_id.0, &str_rev_reg, None)
+            .await?;
+
+        let str_rev_reg_delta = serde_json::to_string(&rev_reg_delta)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_REV_REG_DELTA, &rev_reg_id.0, &str_rev_reg_delta, None)
+            .await?;
+
+        Ok((rev_reg_id.0, str_rev_reg_def, str_rev_reg))
```

Comment:
Obivously the string make it difficult to assure we are returning what calling code expects to be returned in these strings, but I found there's a minor incompatibiltiy on 2nd element `str_rev_reg_def`

While `vdrtools` returned instance of (`libvdrtools/src/domain/anoncreds/revocation_registry_definition.rs`)
```
#[derive(Deserialize, Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RevocationRegistryDefinitionV1 {
    pub id: RevocationRegistryId,
    pub revoc_def_type: RegistryType,
    pub tag: String,
    pub cred_def_id: CredentialDefinitionId,
    pub value: RevocationRegistryDefinitionValue,
}
```
the credx implementation returns instance of 
```
#[derive(Clone, Debug)]
#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
#[cfg_attr(feature = "serde", serde(tag = "ver"))]
pub enum RevocationRegistryDefinition {
    #[cfg_attr(feature = "serde", serde(rename = "1.0"))]
    RevocationRegistryDefinitionV1(RevocationRegistryDefinitionV1),
}

#[derive(Clone, Debug)]
#[cfg_attr(
    feature = "serde",
    derive(Deserialize, Serialize),
    serde(rename_all = "camelCase")
)]
pub struct RevocationRegistryDefinitionV1 {
    pub id: RevocationRegistryId,
    pub revoc_def_type: RegistryType,
    pub tag: String,
    pub cred_def_id: CredentialDefinitionId,
    pub value: RevocationRegistryDefinitionValue,
}
```

Currently I would expect this to cause failure in `generate_rev_reg` (`aries_vcx/src/common/primitives/revocation_registry.rs`) where we try to deserialize this into custom aries-vcx type `RevocationRegistryDefinition` (`aries_vcx/src/common/primitives/revocation_registry.rs:285`)

Reply:
The third element of tuple also check out ✅  - produced json looks as expected
```json
{
    "ver": "1.0",
    "value": {
        "accum": "21 129DC07087DB2972AF5B794DF713B445B8C1B0C5BF157C8FF6F0661A0E17ECF10 21 13862D252ED89BD0522029AB666E6B8296457A3322B4730A7CB0AB93BB15EFB75 6 7F9B8103C80DD5A9C5ABD3FE265027D01B8491C4293DE0C0BC8CDF0AF245309D 4 293A8BF9A068F8F6031B626449307EFF8D83523265CDCED77FFA776BB49B0515 6 613376EC16552DD2ED51594C2EBC1A2E1212E770DB57F3F650293A9FD9BAAEBC 4 2B2DD93B763AEA054EC70ECFFB20AA5222177C5FCD7566A5EF62DA82D1433C82"
    }
}
```

Reply:
Are you sure about this? It looks like they returned `RevocationRegistryDefinition` as well (the parent enum).

Reply:
yeah you are right, the returns on create revocation registry method lgtm ✅